### PR TITLE
feat(head): improved Unhead integration

### PR DESCRIPTION
**Pull request by harlan-zw**
### 🔗 Linked issue

https://github.com/unjs/unhead/issues/116, https://github.com/nuxt/nuxt/issues/15668

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

Unhead [v1.1.0](https://github.com/unjs/unhead/releases/tag/v1.1.0) was recently released.

With the dep update, I figure it's a good time to improve the integration:

## PR To Do

- [ ] benchmark v1.1  
- [ ] test treeshaking

### Features

**Dropping @vueuse/head dependency**

To avoid breaking a bunch of third-party packages, the dependency was kept on @vueuse/head, which acted as a simple compatibility wrapper for Unhead. While we still need some support for this, Unhead now exports the compatibility layer directly. 

This gives finer control over what's imported, allowing for better tree-shaking. For example, `@unhead/dom` should be treeshaken from the server build now.

**Support app.head.options**

While there are minimal build-time options available for Unhead, this is needed to enable `experimentalHashHydration` which I'm going to enable for `nuxt-unhead`. It will be enabled in the future by default but want to test it out a bit first.

**SSR: Skip app.head entry client-side**

This provides a performance improvement for SSR apps, fixes the bug https://github.com/unjs/unhead/issues/116 and allows for `experimentalHashHydration`.

```ts
export default defineNuxtConfig({
  app: {
    head: {
      options: {
        experimentalHashHydration: true
      }
    }
  }
})
```

**New composables**

useServerSeoMeta - Basically what was already available but it provides a `{ mode: 'server' }` which allows for the experimental hash hydration

useHeadSafe / useServerHeadSafe - Bit verbose having both but we need the server function for the above and to treeshake it from the client build (once the vite plugin is implemented)

**Unhead Unplugin**

- Treeshakes server components
- Transforms useSeoMeta (saves ~2kb gzipd)

### Todo (subsequent PRs?)

The following items are things I'd like to solve but I don't think can exist within this PR.

**Expose Unhead Hooks**

These are pretty easy to implement but not sure if this PR is the best spot for them. They're currently not documented within the Unhead docs so should probably sort that out first.

**Document Template Params**

These are mainly used for lower-level integrations at this point.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/nuxt/src/head/runtime/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  let head: VueHeadClient<HeadAugmentations>
+  if (process.server) {
+    head = createServerHead<HeadAugmentations>(headOptions)
+    // when SSR it's safe to only render the appHead server side
+    head.push(appHead, { mode: 'server' })
+  } else {
+    head = createHead<HeadAugmentations>(headOptions)
+    // only in SPA mode do we need to push the appHead client side
+    if (nuxtApp.ssrContext?.noSSR) {
```
Comment (harlan-zw) [inline packages/nuxt/src/head/runtime/plugin.ts:24]: Is this the correct way to check for SPA?

File: packages/nuxt/src/imports/presets.ts [hunk 2]
```diff
@@ -57,7 +54,14 @@ const appPreset = defineUnimportPreset({
     'prefetchComponents',
     'loadPayload',
     'preloadPayload',
-    'isPrerendered'
+    'isPrerendered',
+    // head
```
Comment (harlan-zw) [inline packages/nuxt/src/imports/presets.ts:58]: Wasn't sure why we were importing from `#app`, was there a specific reason?

File: packages/nuxt/src/head/runtime/composables.ts [hunk 3]
```diff
@@ -1,39 +1,57 @@
...
  */
-export const useServerSeoMeta: typeof _useSeoMeta = (meta) => {
+export const useServerSeoMeta: typeof _useServerSeoMeta = (meta) => {
   if (process.server) {
-    return _useSeoMeta(meta)
+    return _useServerSeoMeta(meta)
   }
 }
+
+export {
```
Comment (harlan-zw) [inline packages/nuxt/src/head/runtime/composables.ts:53]: We could wrap these but we don't need to augment the types so not much point.

File: packages/nuxt/src/head/runtime/composables.ts [hunk 4]
```diff
@@ -1,39 +1,57 @@
...
+ * The `useServerHeadSafe` composable is identical to `useHeadSafe` except that it will
+ * have no effect (and will return nothing) if called on the client.
  *
- * It is advised to use `useServerSeoMeta` unless you _need_ client-side
- * rendering of your SEO meta tags.
+ * @see useHeadSafe
  */
-export const useSeoMeta: typeof _useSeoMeta = (meta) => {
-  return _useSeoMeta(meta)
+export const useServerHeadSafe: typeof _useServerHeadSafe = (input, options) => {
```
Comment (harlan-zw) [inline packages/nuxt/src/head/runtime/composables.ts:34]: Both `useServerHeadSafe` and `useServerSeoMeta` can be imported from @unhead/vue when the Vite plugin is implemented

**Comment by danielroe** [comment]
@harlan-zw I think a fair amount of this has been merged in other PRs. But I think there's still some stuff here we would want to merge in, right?

**Comment by harlan-zw** [comment]
Thanks for the ping. I'll close for now and create separate PRs where appropriate.

**Pull request closed without merging**